### PR TITLE
Refactor CUDA Q  Documentation

### PR DESCRIPTION
**Pull request by anthony-santana**
Description

* New Quick Start guide
* New Basics Section
* Revamped Examples and Applications Sections
* New Backends Section
* New Installation Section
* New Extending Section
* Updates all python examples to introduce new python syntax

### CODE REVIEW

File: docs/sphinx/index.rst [hunk 1]
```diff
@@ -3,22 +3,31 @@ CUDA Quantum
...
 
 Welcome to the CUDA Quantum documentation page!
-You are browsing the documentation for |version| version of CUDA Quantum. You can find documentation for all released versions :doc:`here <versions>`.
+
+CUDA Quantum streamlines hybrid application development and promotes productivity and scalability
+in quantum computing. It offers a unified programming model designed for a hybrid
+setting |---| that is, CPUs, GPUs, and QPUs working together. CUDA Quantum contains support for 
+programming in Python and in C++.
+
+You are browsing the documentation for |version| version of CUDA Quantum. You can find documentation for all released versions :doc:`here <releases>`.
```
Comment (bettinaheim) [inline docs/sphinx/index.rst:12]:
```suggestion
You are browsing the documentation for |version| version of CUDA Quantum. You can find documentation for all released versions :doc:`here <versions>`.
```

File: docs/sphinx/using/backends/backends.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
+CUDA Quantum Backends
+**********************
+
+.. toctree::
+   :caption: Backend Targets
+   :maxdepth: 1
+
+      Simulation <simulators.rst>
+      Hardware <hardware.rst>
```
Comment (bettinaheim) [inline docs/sphinx/using/backends/backends.rst:9]:
```suggestion
      Quantum Hardware<hardware.rst>
```

File: docs/sphinx/using/basics/run_kernel.rst [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+  A variety of methods can be used to extract useful information from a :class:`cudaq.SampleResult`. For example,
+  to return the most probable measurement and its respective probability:
+
+  .. literalinclude:: ../../snippets/python/using/first_sample.py
+        :language: python
+        :start-after: [Begin Sample3]
+        :end-before: [End Sample3]
+
+  We can execute this program as we do any python file.
```
Comment (bettinaheim) [inline docs/sphinx/using/basics/run_kernel.rst:38]:
```suggestion
  We can execute this program as we do any Python file.
```

File: docs/sphinx/using/examples/vqe.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
+Variational Quantum Eigensolver
```
Comment (bettinaheim) [inline docs/sphinx/using/examples/vqe.rst:1]: Why does this page not contain a tab for Python/C++? There is a vqe_h2.cpp example to put here.

Comment (anthony-santana) [reply docs/sphinx/using/examples/vqe.rst:1]: I was trying to make all of these examples 1-to-1 and the only C++ VQE examples I could find on hand -- including vqe_h2.cpp -- use different models in their code. I'll link to that one here though and file an issue to write examples to match the python code.

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**